packages/flutter/lib/src/material/icon.dart

```diff
@@ -65,13 +61,26 @@ class Icon extends StatelessComponent {
       category = parts[0];
       subtype = parts[1];
     }
-    String colorSuffix = _getColorSuffix(context);
-    int iconSize = _kIconSize[size];
+    final IconThemeColor iconThemeColor = _getIconThemeColor(context);
+    final String colorSuffix = (iconThemeColor == IconThemeColor.black) ? "black" : "white";
```

Comment:
use a switch when comparing all the values of an enum. That way, when we add another value to the enum later, the analyzer will warn us that we missed one.